src/app/features/reporting/query.service.ts

```diff
@@ -75,35 +120,25 @@ export class QueryService {
     }).value;
   }
 
-  private async loadData(from: Date): Promise<void> {
-    const entityClasses: [EntityConstructor<any>, () => Promise<Entity[]>][] = [
-      [Child, () => this.entityMapper.loadType(Child)],
-      [School, () => this.entityMapper.loadType(School)],
-      [RecurringActivity, () => this.entityMapper.loadType(RecurringActivity)],
-      [
-        ChildSchoolRelation,
-        () => this.entityMapper.loadType(ChildSchoolRelation),
-      ],
-      [Note, () => this.childrenService.getNotesInTimespan(from)],
-      [
-        EventNote,
-        () => this.attendanceService.getEventsOnDate(from, new Date()),
-      ],
-    ];
-
-    this.entities = {};
-
-    const dataPromises = [];
-    for (const entityLoader of entityClasses) {
-      const promise = entityLoader[1]().then((loadedEntities) =>
-        this.setEntities(entityLoader[0], loadedEntities)
-      );
-      dataPromises.push(promise);
+  private async loadEntityCacheIfRequired(query: string, from: Date) {
+    if (!this.entities || Object.keys(this.entities).length === 0) {
+      for (const entityType of this.entityTypesPrimary) {
+        const loadedEntities = await this.entityMapper.loadType(entityType);
+        this.setEntities(entityType, loadedEntities);
+      }
     }
 
-    await Promise.all(dataPromises);
+    if (
+      this.entityTypesTimebound.some((t) => query.includes(t[0].ENTITY_TYPE)) &&
```

Comment:
same filter could be applied to "primary" entities as well

Reply:
But its a bit more challenging as we would also need to check for some (at least one) of the more specific functions: e.g. calling `getAttendanceArray(true)` needs the `ChildSchoolRelation` to be cached.